File: action-type/13-template/go/json_config_loader.go
package main

import (
	"fmt"
)

type JSONConfigLoader struct {
	BaseConfigLoader
}

func (j *JSONConfigLoader) getReader(filePath string) interface{} {
	fmt.Println("Loading JSON config from:", filePath)
	return nil
}

func (j *JSONConfigLoader) parseConfig(reader interface{}) interface{} {
	fmt.Println("Parsing JSON config...")
	return struct{}{}
}

func (j *JSONConfigLoader) validateConfig(config interface{}) bool {
	fmt.Println("Validating JSON config...")
	return true
}

func (j *JSONConfigLoader) Load(filePath string) interface{} {
	return j.BaseConfigLoader.Load(filePath, j.getReader, j.parseConfig, j.validateConfig)
}

File: action-type/13-template/java/JSONConfigLoader.java

package main;

import java.io.Reader;

public class JSONConfigLoader extends AbstractConfigLoader<Object> {

    @Override
    protected Reader getReader(String filePath) {
        System.out.println("Loading JSON config from: " + filePath);
        // 实际实现应返回文件读取器
        return null;
    }

    @Override
    protected Object parseConfig(Reader reader) {
        System.out.println("Parsing JSON config...");
        // 实际实现应解析 JSON 并返回配置对象
        return new Object();
    }
}
